### ISSUE TITLE
Support nonlinear problems?

### ISSUE THREAD

**Issue by KenZhenLin** [JuliaOpt/MosekTools.jl#50]
Hi, I tried to test the exponential cone constraint feature in Mosek 9.1.2 with a simple example: min exp(x) s.t. x>=0 and I use the equivalent formulation: min t s.t. t>=exp(x), x>=0. But I got the error:"The solver does not support nonlinear problems (i.e., NLobjective and NLconstraint)." The code I used is the following:

using JuMP, LinearAlgebra, MosekTools
M = Model(with_optimizer(Mosek.Optimizer))
@variable(M, x)
@variable(M, t)
@NLconstraint(M, t-exp(x)>=0)
@constraint(M, x>=0)
optimize!(M)

Could you please suggest a way out of this? The test was performed in Julia 1.1.1 with JuMP v0.20.1, MosekTools v0.9.3

Thanks!

**Comment by odow** [comment]
You need to use the conic form:
```Julia
using JuMP, MosekTools
model = Model(Mosek.Optimizer)
@variable(model, x >= 0)
@variable(model, t)
@constraint(model, [t, 1, x] in MOI.ExponentialCone())
optimize!(model)
```

Here is a list of cones that you can use: https://www.juliaopt.org/MathOptInterface.jl/stable/apireference/#MathOptInterface.ExponentialCone

**Comment by KenZhenLin** [comment]
> You need to use the conic form:
> 
> ```julia
> using JuMP, MosekTools
> model = Model(Mosek.Optimizer)
> @variable(model, x >= 0)
> @variable(model, t)
> @constraint(model, [t, 1, x] in MOI.ExponentialCone())
> optimize!(model)
> ```
> 
> Here is a list of cones that you can use: https://www.juliaopt.org/MathOptInterface.jl/stable/apireference/#MathOptInterface.ExponentialCone

Thank you! It works. Do you know if we can do something similar with JuMP v0.18.6?  I tried your code and I got "UndefVarError: MOI not defined".

**Comment by odow** [comment]
No. You cannot do this in JuMP 0.18.6. Better support for conic sets was one of the primary motivations for rewriting JuMP.

**Comment by KenZhenLin** [comment]
> No. You cannot do this in JuMP 0.18.6. Better support for conic sets was one of the primary motivations for rewriting JuMP.

Thank you! I'm bit confused because I do see MathProgBase support Exponential cone "https://mathprogbasejl.readthedocs.io/en/latest/conic.html". I just don't know the correct syntax to use it with JuMP 0.18.6.

**Comment by mlubin** [comment]
> I just don't know the correct syntax to use it with JuMP 0.18.6.

As @odow said, there is no syntax to use the exponential cone in JuMP 0.18.6. It was defined in MathProgBase but not exposed through JuMP.
